modules/sample-dropwizard/src/test/scala/core/Dropwizard/DropwizardRoundTripTest.scala

```diff
@@ -0,0 +1,90 @@
+package core.Dropwizard
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import examples.client.dropwizard.user.UserClient
+import examples.server.dropwizard.definitions.User
+import examples.server.dropwizard.user._
+import helpers.MockHelpers._
+import java.util
+import java.util.Optional
+import java.util.concurrent.{CompletableFuture, CompletionStage}
+import org.asynchttpclient.{Request, Response}
+import org.mockito.{ArgumentMatchersSugar, MockitoSugar}
+import org.scalatest.concurrent.Waiters
+import org.scalatest.{FreeSpec, Matchers}
+import scala.compat.java8.FunctionConverters._
+
+class DropwizardRoundTripTest extends FreeSpec with Matchers with Waiters with MockitoSugar with ArgumentMatchersSugar {
+  private implicit val mapper = new ObjectMapper
+
+  "Test server" in {
+    val USERNAME = "foobar"
+
+    val (asyncResponse, serverFuture: CompletableFuture[GetUserByNameResponse]) = mockAsyncResponse[GetUserByNameResponse]
+
+    val resource = new UserResource(new UserHandler {
+      override def createUser(body: User): CompletionStage[CreateUserResponse] = ???
+      override def createUsersWithArrayInput(body: util.List[User]): CompletionStage[CreateUsersWithArrayInputResponse] = ???
+      override def createUsersWithListInput(body: util.List[User]): CompletionStage[CreateUsersWithListInputResponse] = ???
+      override def loginUser(username: String, password: String): CompletionStage[LoginUserResponse] = ???
+      override def logoutUser(): CompletionStage[LogoutUserResponse] = ???
+      override def updateUser(username: String, body: User): CompletionStage[UpdateUserResponse] = ???
+      override def deleteUser(username: String): CompletionStage[DeleteUserResponse] = ???
+
+      override def getUserByName(username: String): CompletionStage[GetUserByNameResponse] = {
+        username match {
+          case USERNAME =>
+            serverFuture.complete(new GetUserByNameResponse.Ok(User.builder()
+              .withEmail("foo@bar.com")
+              .withFirstName("Foo")
+              .withLastName("Bar")
+              .withId(1)
+              .withUsername(USERNAME)
+              .build()
+            ))
+          case "" =>
+            serverFuture.complete(new GetUserByNameResponse.BadRequest)
+          case _ =>
+            serverFuture.complete(new GetUserByNameResponse.NotFound)
+        }
+        serverFuture
+      }
+    })
+
+    val httpClient: Request => CompletionStage[Response] = { request =>
```

Comment:
This kind of test harness looks useful; is this what you were talking about with generating test classes as well?

Reply:
Nah, the test class generation was to ensure that any custom types had a `Showable` implementation, but there was a chicken/egg problem there, and I haven't thought about it more to see if there's anything we can do.